src/main/java/data/TaskManager.java

```diff
@@ -409,33 +408,27 @@ public void markManager(WeekView weekView, MonthView monthView, boolean inMonthV
      * @return The date corresponding to the day number.
      * @throws TaskManagerException If the date is not in the current month or week being viewed.
      */
-    private static LocalDate findDateFromDayNumber(WeekView weekView, MonthView monthView, 
-            boolean inMonthView, int dayInt) throws TaskManagerException {
+    private static LocalDate findDateFromDayNumber(WeekView weekView, MonthView monthView, boolean inMonthView,
+            int dayInt) throws TaskManagerException {
         LocalDate date;
         if (inMonthView) {
-            date = monthView.getStartOfMonth().plusDays(dayInt);
+            LocalDate startOfMonth = monthView.getStartOfMonth();
+            LocalDate endOfMonth = startOfMonth.withDayOfMonth(startOfMonth.lengthOfMonth());
+            if (dayInt < 1 || dayInt > endOfMonth.getDayOfMonth()) {
+                throw new TaskManagerException("Invalid day for month view. Please enter a day between 1 and " +
+                        endOfMonth.getDayOfMonth() + ".");
+            }
```

Comment:
Good that more exception handling has been added to manage cases where the user enters a day that does not exist within the current month